### PR TITLE
Add teamservers

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

This pull request adds teamservers to the COOL assessment environments.  It also moves the EFS mount point for the Kali instances from `/data` to `/share`.

## 💭 Motivation and Context

The COOL assessment environments require teamservers, and the assessment folks have requested the mount point change.

## 🧪 Testing

I successfully deployed this code to staging.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,6 +96,12 @@ the COOL environment.
 |------|-------------|
 | remote_desktop_url | The URL of the remote desktop gateway (Guacamole) for this assessment. |
 
+## Notes ##
+
+Running `pre-commit` requires running `terraform init` in every directory that
+contains Terraform code. In this repository, these are the main directory and
+every directory under `examples/`.
```
Comment (dav3r) [inline README.md]: Should read:

> "In this repository, this is only the main directory."

File: kali_cloud_init.tf [hunk 2]
```diff
@@ -15,7 +15,7 @@ data "template_cloudinit_config" "kali_cloud_init_tasks" {
       "${path.module}/cloud-init/efs-mount.tpl.yml", {
         # Just mount the EFS mount target in the first private subnet
         efs_id      = aws_efs_mount_target.target[var.private_subnet_cidr_blocks[0]].file_system_id
-        mount_point = "/data"
+        mount_point = "/share"
```
Comment (dav3r) [inline kali_cloud_init.tf:18]: I see that the appropriate [corresponding change](https://github.com/cisagov/ansible-role-xfce-cool/pull/2) is coming to `ansible-role-xfce-cool`. 👍

File: teamserver_iam.tf [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  role = aws_iam_role.teamserver_instance_role.name
+}
+
+# The instance role
+resource "aws_iam_role" "teamserver_instance_role" {
+  provider = aws.provisionassessment
+
+  name = "teamserver_instance_role_${terraform.workspace}"
+  # We can just reuse the kali assume role policy here.  If we created
+  # a temserver-specific one it would be identical.
```
Comment (dav3r) [inline teamserver_iam.tf]: Typo: "temserver" -> "teamserver"

File: README.md (hunk 1 above)
Comment (dav3r) [inline README.md:103]: Should read:

> "In this repository, this is only the main directory."

**Review by dav3r — APPROVED** [review]
Excellent work! 🥥

File: teamserver_iam.tf (hunk 3 above)
Comment (jsf9k) [reply teamserver_iam.tf]: See commit 0097fdf.

File: README.md (hunk 1 above)
Comment (jsf9k) [reply README.md]: See commit 9b67e03.

**Review by felddy — APPROVED** [review]
Sharing is caring!  🤟

**Pull request merged**